### PR TITLE
[ml] Switch from `tune.run` to `Tuner.fit`

### PR DESCRIPTION
**Pull request by amogkam**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by xwjiang2010 — APPROVED** [review]

**Review by Yard1 — APPROVED** [review]

File: python/ray/tune/impl/tuner_internal.py [hunk 1]
```diff
@@ -84,14 +84,14 @@ def __init__(
 
         self._is_restored = False
         self._trainable = trainable
-        self._tune_config = tune_config
-        self._run_config = run_config
+        self._tune_config = tune_config if tune_config else TuneConfig()
+        self._run_config = run_config if run_config else RunConfig()
```
Comment (Yard1) [inline python/ray/tune/impl/tuner_internal.py:88]:
```suggestion
        self._tune_config = tune_config or TuneConfig()
        self._run_config = run_config or RunConfig()
```

File: python/ray/tune/impl/tuner_internal.py [hunk 2]
```diff
@@ -84,14 +84,14 @@ def __init__(
...
         self._experiment_checkpoint_dir = self._setup_create_experiment_checkpoint_dir(
             self._run_config
         )
 
         # Not used for restored Tuner.
-        self._param_space = param_space
+        self._param_space = param_space if param_space else {}
```
Comment (Yard1) [inline python/ray/tune/impl/tuner_internal.py:94]:
```suggestion
        self._param_space = param_space or {}
```

**Review by Yard1 — COMMENTED** [review]
More concise:

**Comment by amogkam** [comment]
FYI @krfricke looks like the tests are failing because Tuner.fit returns a `ray.tune.Checkpoint` not a `ray.ml.Checkpoint`.

**Comment by amogkam** [comment]
Blocked by https://github.com/ray-project/ray/issues/23295

**Pull request merged**